openedx/adg/lms/applications/helpers.py

```diff
@@ -285,3 +302,127 @@ def has_admin_permissions(user):
+                is_any_course_started = True
+                status = 'In Progress'
+                course_grade = CourseGradeFactory().read(user, course_key=course.id)
+
+                if course_grade:
+                    if course_grade.passed:
+                        status = 'Completed'
+                        grade = str(int(round_away_from_zero(course_grade.percent * 100)))
+                    elif has_attempted_all_modules(user, course):
+                        status = 'Re-Take Course'
+                        message = RETAKE_COURSE_MESSAGE
+                        grade = str(int(round_away_from_zero(course_grade.percent * 100)))
+            else:
+                status = 'Not Started'
+
+        course_cards_information.append({
+            'course': course,
+            'status': status,
+            'grade': grade,
+            'message': message
+        })
+
+    return course_cards_information, is_any_course_started, is_locked
+
+
+def has_attempted_all_modules(user, course):
+    """
+    Checks if all sections of a course have been attempted by the user or not
+
+    Arguments:
+        user: User object
+        course: Course object
+
+    Returns:
+        boolean: True if user attempted all sections, False otherwise
+    """
+    all_modules = modulestore().get_items(
+        course.id,
+        qualifiers={'category': 'course'}
+    )
+
+    for module in all_modules[0].children:
+        if not StudentModule.objects.filter(student=user, module_state_key=module).exists():
+            return False
+
+    return True
+
+
+def get_application_hub_instructions(user_application_hub, is_any_prerequisite_started, is_any_bl_course_started):
+    """
+    Decides congratulation messages and instructions based on the requirements completed and if courses have been
+    started or not.
+
+    Arguments:
+        user_application_hub: ApplicationHub object
+        is_any_prerequisite_started: boolean
+        is_any_bl_course_started: boolean
+
+    Returns:
+        dictionary containing congratulations messages and instructions
+    """
+    congrats = message = instruction = ''
+
+    if user_application_hub.are_application_pre_reqs_completed():
+        congrats = APPLICATION_SUBMISSION_CONGRATS
+        message = APPLICATION_SUBMISSION_INSTRUCTION
+    elif user_application_hub.is_prerequisite_courses_passed:
+        if is_any_bl_course_started:
+            instruction = PREREQUISITE_COURSES_COMPLETION_INSTRUCTION
+        else:
+            congrats = PREREQUISITE_COURSES_COMPLETION_CONGRATS
+            message = PREREQUISITE_COURSES_COMPLETION_MSG
+    elif user_application_hub.is_written_application_completed:
+        if is_any_prerequisite_started:
+            instruction = WRITTEN_APPLICATION_COMPLETION_INSTRUCTION
+        else:
+            congrats = WRITTEN_APPLICATION_COMPLETION_CONGRATS
+            message = WRITTEN_APPLICATION_COMPLETION_MSG
```

Comment:
Can we add some line spaces between different conditions to enhance readability?